src/Tribe/Editor/Hooks.php

```diff
@@ -27,6 +30,76 @@ public function register() {
 	 */
 	protected function add_actions() {
 		add_action( 'current_screen', [ $this, 'add_widget_resources' ] );
+		add_action( 'save_post_' . TEC::POSTTYPE, [ $this, 'calculate_duration' ], 20 );
+		add_action( 'plugins_loaded', [ $this, 'update_old_durations' ] );
+	}
+
+	/**
+	 * Calculate the event duration based on the saved start and end dates.
+	 * This occurs during post save, and checks the saved value against the calculated value on updates.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $post_id The event we are setting a duration for.
+	 */
+	public function calculate_duration( $post_id ) {
+		// Sanity check.
+		if ( TEC::POSTTYPE !== get_post_type( $post_id ) ) {
+			return;
+		}
+
+		$saved_duration = get_post_meta( $post_id, '_EventDuration', true  );
+		$start_date_utc = get_post_meta( $post_id, '_EventStartDateUTC', true );
+		$end_date_utc   = get_post_meta( $post_id, '_EventEndDateUTC', true  );
+
+		// Don't try to calculate if a date is missing.
+		if ( empty( $start_date_utc ) || empty( $end_date_utc ) ) {
+			return;
+		}
+
+		$utc_timezone          = new \DateTimezone( 'UTC' );
+		$start_date_utc_object = Dates::immutable( $start_date_utc, $utc_timezone );
+		$end_date_utc_object   = Dates::immutable( $end_date_utc, $utc_timezone );
+		$calculated_duration   = $end_date_utc_object->getTimestamp() - $start_date_utc_object->getTimestamp();
+
+		if ( empty( $saved_duration ) || (int) $saved_duration !== (int) $calculated_duration ) {
```

Comment:
This check should be done before, right after you fetch the `$saved_duration` value and avoid all the wasted computation that got you here.

Reply:
I mean the `empty( $saved_duration )` check.

Reply:
No.
I create an event. The duration is calculated.
I then go in and change the dates - the meta exists but the duration needs to be recalculated.